contracts/Lottery.sol

```diff
@@ -93,54 +125,56 @@ contract Lottery {
         
         uint randomNumber = uint(keccak256(abi.encodePacked(now,  tickets.length))) % tickets.length;
         LotteryTicket memory winnerTicket = tickets[randomNumber - 1];
-        winnerTicket.ticketOwner.transfer(address(this).balance);
-        winner.push(playerMap[winnerTicket.ticketOwner]);
         
+        winner = playerMap[winnerTicket.ticketOwner];
+        winner.playerAddress.transfer(getPrizeAmount());
+        
+        state = LotteryState.Finished;
         //Send notification
+        emit LotteryWinner(winner.playerAddress, getPrizeAmount());
+        return winner;
```

Comment:
Is it necessary to return the winner if the event for the same is already emitted?